src/delta.c

```diff
@@ -38,86 +38,165 @@
 #include "swupd.h"
 #include "xattrs.h"
 
-static void do_delta(char *filename, struct file *file);
-
-void try_delta(struct file *file)
+static bool compute_hash_from_file(char *filename, char *hash)
 {
-	char *filename;
-	struct stat stat;
+	/* TODO: implement this without the indirection of creating a file... */
+	struct file f;
 
-	if (file->is_file == 0) {
-		return;
-	}
+	f.filename = filename;
+	f.use_xattrs = true;
```

Comment:
If I recall correctly, I've pulled this logic from verify_file. The goal was to perform the same hash calculation as a verify would.

Reply:
I was curious about this too when I reviewed so I looked at other hash calculations in the code to make sure it was done the same. I'm ok with setting use_xattrs to be true even though we don't expect any at least for consistency. We can remove that assignment in the future after some testing if it is necessary.

Also, didn't yocto/meta-swupd use xattrs? So we need to use_xattrs for other use-cases. If none exist it shouldn't change our calculation anyways.

Reply:
There is a configure-time option (--enable-xattrs) to enable xattr support. It defaults to disabled...

Reply:
The configure time option controls whether the querying for xattrs is real or just give dummy answers ("no xattrs!"). See xattrs.c:41. Setting use_xattrs won't cause reading actual xattrs unless we have it enabled.

It might be worthwhile to revisit how xattrs is enabled/disabled, but for this patch doing the same thing as verify_file is the right call.